charts/kong/dispatch-transformer/handler.lua

```diff
@@ -84,15 +89,34 @@ local function substitute_payload(conf, result)
   if data then
     ngx.log(ngx.DEBUG, "request body: " .. data)
   else
-    data = "{}"
+    data = {}
   end
 
-  local ok, json_data = parse_json(data)
-  if ok then
-    ngx.log(ngx.DEBUG, "request body is json")
-    result[conf.substitute.input] = json_data
+  local header = ngx.req.get_headers()
+  if is_json_body(header) then
+    local ok, json_data = parse_json(data)
+    if ok then
+      ngx.log(ngx.DEBUG, "request body is json")
+      result[conf.substitute.input] = json_data
+    else
+      ngx.log(ngx.DEBUG, "request body is not json")
+      result[conf.substitute.input] = data
```

Comment:
This seems wrong.  Shouldn't we return an HTTP error if the content type is application/json, but the payload doesn't match that?

Reply:
I guess that's just continuing existing behavior, but it seems wrong

Reply:
Yeah that's a good point. I was just extending the existing behavior. I'll look into throwing an HTTP error when the payload doesn't match the content type or when the content type is not supported or specified.